src/redisearch_rs/inverted_index/src/tests.rs

```diff
@@ -313,18 +313,21 @@ fn reading_records() {
         .expect("to be able to read from the buffer")
         .expect("to get a record");
     assert_eq!(record, RSIndexResult::virt().doc_id(10));
+    drop(record);
```

Comment:
Why are these needed?

Reply:
We need to drop to release the borrow.

```
error[E0499]: cannot borrow `ir` as mutable more than once at a time
   --> inverted_index/src/tests.rs:317:18
    |
311 |     let record = ir
    |                  -- first mutable borrow occurs here
...
317 |     let record = ir
    |                  ^^ second mutable borrow occurs here
...
335 | }
    | - first borrow might be used here, when `record` is dropped and runs the `Drop` code for type `RSIndexResult`
    ```

Reply:
Yes, that does not help.

I think that's a limitation of the borrow checker not considering that the previous binding is drop before re-calling the method.